Push/PushCrypto.swift

```diff
@@ -207,10 +313,18 @@ extension String {
 
 extension Data {
     /// Returns a base64 url safe encoding of the given data.
-    var base64urlSafeEncodedString: String {
-        return self.base64EncodedString
-            .replacingOccurrences(of: "+", with: "-", options: .literal)
-            .replacingOccurrences(of: "/", with: "_", options: .literal)
-            .replacingOccurrences(of: "=", with: "", options: [.anchored, .backwards])
+    var base64urlSafeEncodedString: String? {
+        let length = ece_base64url_encode(self.getBytes(), self.count, ECE_BASE64URL_OMIT_PADDING, nil, 0)
+        guard length > 0 else {
+            return nil
+        }
+
+        var bytes = [CChar](repeating: 0, count: length)
+        let checkLength = ece_base64url_encode(self.getBytes(), self.count, ECE_BASE64URL_OMIT_PADDING, &bytes, length)
+        guard checkLength == length else {
+            return nil
+        }
+
+        return String(data: Data(bytes: bytes, count: length), encoding: String.Encoding.ascii)
```

Comment:
Nice, I didn't know about the `.ascii` shorthand. IIUC, Base64 should only ever be (7-bit) ASCII, but I can change it to `.utf8` if you'd like.